### ISSUE TITLE
Proposal - jax.jit static_argnums and static_argnames raises warning if parameter doesn't exist

### ISSUE THREAD

**Issue by pseudo-rnd-thoughts** [google/jax#10046]
Currently, if a static_argnums or static_argname doesn't exist for the function, nothing happens
This can be an issue if a function is updated but the jax.jit not being updated

```
import jax
import functools

@functools.partial(jax.jit, static_argnames=('b', 'd'))
def f(a, b, c):
    return a + b + c

print(f(1., 2., 3.))

@functools.partial(jax.jit, static_argnums=(1, 3))
def g(a, b, c):
    return a + b + c

print(g(1., 2., 3.))
```

I propose that a warning is raised if an arg num or arg name doesn't exist such that code will still run if the arg num or name doesn't exist but the programmer is made aware of the issue (and can silence if necessary)
Is there a good reason for this not existing already? 

[x] I can't find a previous issue on this

**Comment by jakevdp** [comment]
One reason this isn't done is that we don't always have the ability to inspect which keyword arguments the function accepts. For example, what would you hope would happen in this case?
```python
@functools.partial(jax.jit, static_argnames=('b', 'd'))
def f(**kwds):
    return kwds['a'] + kwds['b']
```

**Comment by shoyer** [comment]
On the other hand, we could probably safely do this for cases that don't use `*args` or `**kwargs`.

**Comment by pseudo-rnd-thoughts** [comment]
@jakevdp Thanks for the comment, I had considered that case but thought that the whole of `kwargs` would be static not the parts inside. But testing, your right that this propose wouldn't work for those cases. 

However, as @shoyer says, I still think that it is a good idea as it can prevent a number of minor problems in problems code. 
A possible work around is to silence the warning if the *args or **kwargs exists? 
Or to have an extra parameter to silence the warning, however this seems like a worse way of doing it

**Comment by JeppeKlitgaard** [comment]
I only just found this, so I realise I made a duplicate issue over in #10601.

This issue is fixed by #10603. See PR for comments on `*args`, and `**kwargs` cases.
